addons/website_livechat/static/src/models/visitor/visitor.js

```diff
@@ -0,0 +1,143 @@
+                data2.lang = data.lang;
+            }
+            if ('name' in data) {
+                data2.name = data.name;
+            }
+            if ('partner_id' in data) {
+                if (data.partner_id) {
+                    data2.partner = [['insert', { id: data.partner_id }]];
+                } else {
+                    data2.partner = [['unlink-all']];
+                }
+            }
+            if ('website' in data) {
+                data2.website = data.website;
+            }
+            return data2;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeAvatarUrl() {
+            if (!this.partner) {
+                return '/mail/static/src/img/smiley/avatar.jpg';
+            }
+            return this.partner.avatarUrl;
+        }
+
+        /**
+         * @private
+         * @returns {mail.country}
+         */
+        _computeCountry() {
+            if (this.partner && this.partner.country) {
+                return [['link', this.partner.country]];
+            }
+            if (this.country) {
+                return [['link', this.country]];
+            }
+            return [['unlink']];
+        }
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeNameOrDisplayName() {
+            if (this.partner) {
+                return this.partner.nameOrDisplayName;
+            }
+            return this.name;
+        }
+    }
+
+    Visitor.fields = {
+        avatarUrl: attr({
+            compute: '_computeAvatarUrl',
+            dependencies: [
+                'partner',
+                'partnerAvatarUrl'
+            ]
+        }),
+        country: many2one('mail.country', {
+            compute: '_computeCountry',
+            dependencies: [
+                'partnerCountry',
+            ]
+        }),
+        history: attr(),
+        is_connected: attr(),
+        /**
+         * Name of the language of the visitor.
+         */
+        lang: attr(),
+        name: attr(),
+        nameOrDisplayName: attr({
+            compute: '_computeNameOrDisplayName',
+            dependencies: [
+                'name',
+                'partnerNameOrDisplayName',
+            ],
+        }),
+        partner: many2one('mail.partner'),
+        partnerAvatarUrl: attr({
+            related: 'partner.avatarUrl',
+        }),
+        partnerCountry: many2one('mail.country',{
+            related: 'partner.country',
+        }),
+        partnerNameOrDisplayName: attr({related: 'partner.nameOrDisplayName'}),
+        threads: one2many('mail.thread', {
+            inverse: 'visitor',
+        }),
+        /**
+         * Name of the website on which the visitor is connected.
+         */
+        website: attr(),
```

Comment:
_"Name of the website"_? Not super clear, would provide an example similarly to suggestion with `lang` field.